### PR TITLE
Add default expansion.

### LINKED ISSUE #11: Setting default values if not present in environment
Are you able to set default values for the environment variables if they are not present in your environment already, like [in Bash](https://jaduks.livejournal.com/7934.html)?

Or do you recommend we handle this within our code?

### PR DESCRIPTION
**Pull request by vantreeseba**
Allow expansion to occur, but replace non existing with default if
given.

fixes #11 

Example: ${MACHINE-localhost}

Should resolve to $MACHINE expansion, and if not set, then expands to
localhost.

### CODE REVIEW

**Comment by vantreeseba** [comment]
There are new tests for all conditions, let me know if there is anything else you would like from me for this to get in.

**Comment by vantreeseba** [comment]
This should be good to, let me know if there is anything else you feel like should be tested / etc.

**Comment by snyamathi** [comment]
Would using `:-` make more sense to mimic the shell syntax?

```
${parameter:-word}
Use Default Values. If parameter is unset or null, the expansion of word shall be substituted; otherwise, the value of parameter shall be substituted.
```

https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02

**Comment by vantreeseba** [comment]
> Would using `:-` make more sense to mimic the shell syntax?
> 
> ```
> ${parameter:-word}
> Use Default Values. If parameter is unset or null, the expansion of word shall be substituted; otherwise, the value of parameter shall be substituted.
> ```
> 
> https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02

Yeah, that makes more sense, I guess I missed that somehow, and in bash (at least 4.4.20) it works without the ":", so ${non_existing-default} will return default.

I'll look into changing it.

**Comment by paulrostorp** [comment]
any updates on this ?

**Comment by vantreeseba** [comment]
Hey sorry, I honestly completely forgot about this, I will try to look into it this week.

**Comment by vantreeseba** [comment]
With those two latest commits, this should be good to go.
Tests all are passing, let me know if there is anything else you would like.

**Comment by DrunkenPoney** [comment]
Any updates?

**Comment by FezVrasta** [comment]
Anything holding back on this PR?

**Comment by motdotla** [comment]
Great feature. Thank you. Merging and this will be a major upgrade release shortly.

**Comment by motdotla** [comment]
Thank you 💛. This has been released: https://github.com/motdotla/dotenv-expand/blob/master/CHANGELOG.md#600-2022-01-17

File: test/.env [hunk 1]
```diff
@@ -5,6 +5,12 @@ MACHINE=machine_env
 MACHINE_EXPAND=$MACHINE
 UNDEFINED_EXPAND=$UNDEFINED_ENV_KEY
 ESCAPED_EXPAND=\$ESCAPED
+DEFINED_EXPAND_WITH_DEFAULT=${MACHINE:-default}
+DEFINED_EXPAND_WITH_DEFAULT_NESTED=${MACHINE:-${UNDEFINED_ENV_KEY:-default}}
+UNDEFINED_EXPAND_WITH_DEFINED_NESTED=${UNDEFINED_ENV_KEY:-${MACHINE:-default}}
+UNDEFINED_EXPAND_WITH_DEFAULT=${UNDEFINED_ENV_KEY:-default}
+UNDEFINED_EXPAND_WITH_DEFAULT_NESTED=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2:-default}}
+UNDEFINED_EXPAND_WITH_DEFAULT_NESTED_TWICE=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2${UNDEFINED_ENV_KEY_3:-default}}}
```
Comment (FezVrasta) [inline test/.env:13]: @vantreeseba may you help me understand how is this supposed to work? Why are `UNDEFINED_ENV_KEY2` and `${UNDEFINED_ENV_KEY_3:-default}` being concatenated?

Comment (vantreeseba) [reply test/.env:13]: That actually looks like my test was missing a :-

So it should look like
```bash
UNDEFINED_EXPAND_WITH_DEFAULT_NESTED_TWICE=${UNDEFINED_ENV_KEY:-${UNDEFINED_ENV_KEY_2:-${UNDEFINED_ENV_KEY_3:-default}}}
```

Idea being, it's a n, n+1,n+2 test to do a "proof by induction".

I can fix it, or if you want to fix it in your branch is fine, whichever everyone thinks will be easier/better.

**Pull request merged**